server/src/main/java/org/elasticsearch/search/internal/FieldUsageTrackingDirectoryReader.java

```diff
@@ -234,21 +234,19 @@ public ByteVectorValues getByteVectorValues(String field) throws IOException {
         }
 
         @Override
-        public TopDocs searchNearestVectors(String field, byte[] target, int k, Bits acceptDocs, int visitedLimit) throws IOException {
-            TopDocs topDocs = super.searchNearestVectors(field, target, k, acceptDocs, visitedLimit);
-            if (topDocs != null) {
+        public void searchNearestVectors(String field, byte[] target, KnnCollector collector, Bits acceptDocs) throws IOException {
+            super.searchNearestVectors(field, target, collector, acceptDocs);
+            if (collector.visitedCount() > 0) {
```

Comment:
This is the biggest change here. If the collector doesn't visit any documents, that means there weren't matching the `acceptDocs` (match_none), there aren't any docs with that field, or the field just isn't there. 

I am not sure the usage of `FieldUsageTrackingDirectoryReader` and how we need to handle this :/

Reply:
This changes the behavior slightly, before even if we didn't match anything, we would track the field usage. 

Also, previously, if we used `byte[]` to query a `float[]` field, it returned `null`. But, honestly, that seems like a bug to me :/.

Reply:
If we want to keep the behaviour the same I think we can call `super.searchNearestVectors` with a wrapped `KnnCollector` that calls `notifier.onKnnVectorsUsed(field)` in its `collect()` method

Reply:
Boom! 100% you are correct. I will do this.